README.md

```diff
@@ -91,97 +103,151 @@ cloning over SSH may provide a better experience (which requires
+xcodebuild integrates nicely with Xcode.app, but Ninja is a bit faster and
+supports more environments.
 
-    git clone git@github.com:ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+To build using Xcode, run:
 
-#### Install via third-party packaging tool (macOS only)
+    utils/build-script -x
 
-**[Homebrew](https://brew.sh/)**
+The Xcode IDE can be used to edit the Swift source code, but it is not currently
+fully supported as a build environment for SDKs other than macOS. If you need to
+work with other SDKs, you'll need to create a second build using Ninja.
 
-    brew install cmake ninja
+To build using Ninja, run:
 
-**[MacPorts](https://macports.org)**
+    utils/build-script -R --debug-swift
 
-    sudo port install cmake ninja
+This will build LLVM+Clang with optimizations and no debug symbols, but will
+build swift with debug symbols. This is a good starting point since the debugger
+will work in the swift code base, and the tests will run faster.
 
-### Building Swift
+#### Build Products
 
-The `build-script` is a high-level build automation script that supports basic
-options such as building a Swift-compatible LLDB, building the Swift Package
-Manager, building for iOS, running tests after builds, and more. It also
-supports presets, which you can define for common combinations of build options.
+All of the build products are placed in `swift-source/build/${TOOL}-${MODE}/${PRODUCT}-${PLATFORM}/`.
+If macOS swift with Ninja in DebugAssert mode was built, all of the products
+would be in `swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64/`. It
+helps to save this directory as an environment variable for future use.
 
-To find out more:
+    export SWIFT_BUILD_ROOT="~/swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64"
 
-    utils/build-script -h
+#### Ninja
 
-Note: Arguments after "--" above are forwarded to `build-script-impl`, which is
-the ultimate shell script that invokes the actual build and test commands.
+Once the first build has been finished with `build-script`, ninja can perform
+fast incremental builds of various products.
 
-A basic command to build Swift with optimizations and run basic tests with
-Ninja:
+    cd ${SWIFT_BUILD_ROOT}
+    ninja swift
+    ninja swift-stdlib
 
-    utils/build-script -r -t
+These incremental builds are a big timesaver when developing and debugging. It
+is still always a good idea to do a full build after using `update-checkout`.
+Building the `swift-stdlib` target as an additional layer of testing from time
+to time is also a good idea.
 
-## Developing Swift in Xcode
+#### Xcode
 
-`build-script` can also generate Xcode projects:
+To open the swift in Xcode, open `${SWIFT_BUILD_ROOT}/Swift.xcodeproj`. It will
+auto-create a *lot* of schemes for all of the available targets. A common debug
+flow would involve:
 
-    utils/build-script -x
+ - Select the ‘swift’ scheme
+ - Pull up the scheme editor (⌘+⇧+<)
```

Comment:
The ⌘ symbol, and maybe the ⇧ symbol, won't render on Linux,. Might want to spell this as "Command-Shift-<", or as Command-Shift-, or Command-Shift-Comma. Or do it like Xcode does it: Command-<. There's no great answer here :\